src/Symfony/Component/Routing/Requirement/EnumRequirement.php

```diff
@@ -22,12 +22,13 @@ final class EnumRequirement implements \Stringable
 
     /**
      * @template T of \BackedEnum
+     *
      * @param class-string<T> $enum
-     * @param T ...$cases
+     * @param list<T>         $cases
      */
-    public function __construct(string $enum, \BackedEnum ...$cases)
+    public function __construct(string $enum, array $cases = [])
```

Comment:
`\get_class($case)` won't work when `$case` is not an object. We need to add a test covering this new possibility (`$cases` containing a scalar for example).

Reply:
just change the code to use `get_debug_type` for the error message, and avoid accessing `$case->name` when it is not an enum.

Reply:
test added